src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -572,14 +572,57 @@ bool HandleRedirect (HttpStatusCode redirectCode, HttpURLConnection httpConnecti
 			if (redirectState.RedirectCounter >= MaxAutomaticRedirections)
 				throw new WebException ($"Maximum automatic redirections exceeded (allowed {MaxAutomaticRedirections}, redirected {redirectState.RedirectCounter} times)");
 
-			string redirectUrl = locationHeader [0];
-			string protocol = httpConnection.URL?.Protocol;
-			if (redirectUrl.StartsWith ("//", StringComparison.Ordinal)) {
-				// When redirecting to an URL without protocol, we use the protocol of previous request
-				// See https://tools.ietf.org/html/rfc3986#section-5 (example in section 5.4)
-				redirectUrl = protocol + ":" + redirectUrl;
-			}
-			redirectState.NewUrl = new Uri (redirectUrl, UriKind.Absolute);
+			Uri redirectUrl;
+			try {
+				if (Logger.LogNet)
+					Logger.Log (LogLevel.Debug, LOG_APP, $"Raw redirect location: {locationHeader [0]}");
```

Comment:
Nope. The source URI will either be the previous redirect, and that's already in the log, or it will be the original request URI as given by the user. If we want to include the original URI we can add a log line in `SendAsync`, but not here.